File: qqbot-module-onebot11/src/main/java/com/mieai/qqbot/onebot11/mapping/OneBotStoredMessage.java
package com.mieai.qqbot.onebot11.mapping;

import com.mieai.qqbot.client.QqMessageTargetType;
import com.mieai.qqbot.domain.bot.BotId;

public record OneBotStoredMessage(
        int messageId,
        BotId botId,
        String officialMessageId,
        QqMessageTargetType targetType,
        String targetRawId,
        Direction direction,
        String messageType,
        long eventTime,
        long userId,
        String messageJson,
        String senderJson) {

    public enum Direction {
        INCOMING,
        OUTGOING
    }
}

File: qqbot-module-onebot11/src/main/kotlin/com/mieai/qqbot/onebot11/mapping/OneBotStoredMessage.kt
package com.mieai.qqbot.onebot11.mapping

import com.mieai.qqbot.client.QqMessageTargetType
import com.mieai.qqbot.domain.bot.BotId

data class OneBotStoredMessage(
    val messageId: Int,
    val botId: BotId,
    val officialMessageId: String,
    val targetType: QqMessageTargetType,
    val targetRawId: String,
    val direction: Direction,
    val messageType: String,
    val eventTime: Long,
    val userId: Long,
    val messageJson: String,
    val senderJson: String,
) {
    enum class Direction {
        INCOMING,
        OUTGOING,
    }
}
